CI/windows/build_cmocka.ps1

```diff
@@ -0,0 +1,77 @@
+Param(
+    [Switch]$Help = $(if (Test-Path variable:Help) { $Help }),
+    [Switch]$Quiet = $(if (Test-Path variable:Quiet) { $Quiet }),
+    [Switch]$Verbose = $(if (Test-Path variable:Verbose) { $Verbose }),
+    [Switch]$NoChoco = $(if (Test-Path variable:NoChoco) { $NoChoco }),
+    [Switch]$SkipDependencyChecks = $(if (Test-Path variable:SkipDependencyChecks) { $SkipDependencyChecks }),
+    [Switch]$Install = $(if (Test-Path variable:Install) { $Install }),
+    [String]$BuildDirectory = "build",
+    [ValidateSet("32-bit", "64-bit")]
+    [String]$BuildArch = (Get-CimInstance CIM_OperatingSystem).OSArchitecture,
+    [ValidateSet("Release", "RelWithDebInfo", "MinSizeRel", "Debug")]
+    [String]$BuildConfiguration = "RelWithDebInfo"
+)
+
+################################################################################
+# Windows cmocka native-compile build script
+################################################################################
+#
+# This script file can be included in build scripts for Windows or run directly
+#
+################################################################################
+
+function Build-Product {
+    cd "${DepsBuildDir}"
+
+    if ($Quiet) {
+        $CMAKE_OPTS = "-Wno-deprecated -Wno-dev --log-level=ERROR"
+    } else {
+        $CMAKE_OPTS = ""
+    }
+
+    Write-Step "Configure (${ARCH})..."
+    cmake -G "Visual Studio 16 2019" `
+        -A "${CMAKE_ARCH}" `
+        -DBUILD_TESTING=OFF `
+        -DWITH_EXAMPLES=OFF `
+        "${CMAKE_OPTS}" `
+        -S "cmocka" `
+        -B "cmocka_build\${CMAKE_BITNESS}"
+
+    Write-Step "Build (${ARCH})..."
+    cmake --build "cmocka_build\${CMAKE_BITNESS}" --config "${BuildConfiguration}"
+}
+
+function Install-Product {
+    cd "${DepsBuildDir}"
+
+    Write-Step "Install (${ARCH})..."
+    cmake --install "cmocka_build\${CMAKE_BITNESS}" --config "${BuildConfiguration}" --prefix "${DepsBuildDir}\${CMAKE_INSTALL_DIR}"
```

Comment:
Install prefix should've been set at configuration time via `-DCMAKE_INSTALL_PREFIX="[WHEREVER]"` already, but maybe matter of styling.

Reply:
I believe this is where I picked up this pattern:  https://alexreinking.com/blog/how-to-use-cmake-without-the-agonizing-pain-part-2.html

Reply:
Yeah it's canonical, the parameter is there for a reason; my expectation would've been that setting that prior to running `install` avoids another CMake configuration step (as it doesn't have to generate all the `cmake_install.cmake` files, but that's just an optimisation.

Reply:
I'm honestly not sure what the best practice here is.  I have a hazy recollection that doing it this way avoided some issue, but I may be misremembering.  I'll make a note to look into this further.